### ISSUE TITLE
[STD/FS] createSymlink creates invalid symlink to another directory

### ISSUE THREAD

**Issue by ilovewine** [denoland/deno#9225]
**ENV**: Linux

**Steps to reproduce**:
- go to ./folder (in terminal)
- create ./folderTarget/target folder inside ./folder (e.g. mkdir ./folderTarget/target)
- create the script symlink.ts inside ./folder:
```
import {
  ensureSymlink,
} from "https://deno.land/std@0.80.0/fs/mod.ts";

ensureSymlink('./folderTarget/target','./folder/foo');
```
- run deno run --unstable --allow-read --allow-write symlink.ts
**Expected result**: the symlink ./folder/foo is an appropriate symlink and it leads to the target folder
**Actual result**: the symlink is broken (i.e. it doesn't lead to the target folder)

**Comment by ilovewine** [comment]
#9224 - This is an equivalent for a synchronous method.

**Comment by pevers** [comment]
I was looking into this and found out that it happens because the symlinks have relative paths. We should canonicalize them: https://doc.rust-lang.org/std/fs/fn.canonicalize.html to fix it.

I'll open a PR.

**Comment by pevers** [comment]
I'm not sure if this is a bug on second thoughts. If we stick close to the OS:

Doing:
```shell
$ mkdir folder
$ ln -s ./folderTarget/target folder/foo
```

Would have the same results (a broken symlink). 

fs-extra implements a 'smart' strategy: https://github.com/jprichardson/node-fs-extra/blob/6bffcd81881ae474d3d1765be7dd389b5edfd0e0/lib/ensure/symlink-paths.js

It will check if the path is relative to the current working directory. That could work for Deno as well. OR, we add it to the docs that the symlink is relative to the link's parent directory as Node did: https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback .

**Comment by caspervonb** [comment]
Working as intended IMO, your paths are just messed up.

**Comment by lucacasonato** [comment]
This is not a bug. Behavior is intended.

**Comment by caspervonb** [comment]
To expand on this a little bit; this is a system call binding and as such it behaves like the system call.

> Symbolic links are interpreted at run time as if the contents of
> the link had been substituted into the path being followed to find a file or directory.
>
> Symbolic links may contain ..  path components, which (if used at
> the start of the link) refer to the parent directories of that in
> which the link resides.
>
> A symbolic link (also known as a soft link) may point to an
> existing file or to a nonexistent one; the latter case is known
> as a dangling link.

See https://man7.org/linux/man-pages/man2/symlink.2.html for more details.

**Comment by Spoonbender** [comment]
Yup, `std::fs::canonicalize` turns relative path into absolute path, but relative symlinks are actually a feature. They enable, for example, links within some directory and subdirectories structure (e.g. from /examples to /src/example) that remain intact even if the structure is moved elsewhere in the filesystem.
